src/events/messageCreate.ts

```diff
@@ -1,32 +1,33 @@
 import { ChannelType, GuildMemberRoleManager, Message, TextChannel, Webhook, WebhookType } from 'discord.js';
-import { HypixelAPIKeyRegex, IPAddressPattern, URLRegex, DiscordInviteRegex } from '../utils/regex';
 import { autoModBypassRole } from '../../config.json';
 import Infraction from '../utils/Infraction';
+import { DiscordInviteRegex, HypixelAPIKeyRegex, IPAddressPattern, URLRegex } from '../utils/regex';
 
 const allowedDomains: string[] = [
-  'hypixel.net',
   '*.hypixel.net',
-  'discord.com',
   '*.discord.com',
-  'kath.lol',
   '*.kathund.wtf',
-  'kathund.wtf',
+  'kath.lol',
   'hypixel-api-reborn.github.io'
 ];
 
-function matchWildcard(domain: string, pattern: string): boolean {
-  const regexPattern = pattern.replace(/[-\/\\^$+?.()|[\]{}]/g, '\\$&').replace(/\*/g, '.*');
-  const regex = new RegExp(`^${regexPattern}$`);
-  return regex.test(domain);
-}
-
 function isUrlAllowed(url: string): boolean {
   const isValidUrl = URLRegex.test(url);
   if (!isValidUrl) return false;
   const match = url.match(URLRegex);
-  const domain = match ? match[2] : null;
-  if (!domain) return false;
-  return !allowedDomains.some((pattern) => matchWildcard(domain, pattern));
+  if (!match) return false;
+  const domain: string = match[3];
+
+  if (allowedDomains.some((pattern) => pattern === domain) && !match[2]) {
+    return true;
+  } else if (!allowedDomains.some((pattern) => pattern === domain)) {
+    return (
+      allowedDomains.some((pattern) => pattern === `*.${domain}`) ||
+      allowedDomains.some((pattern) => pattern === match[2] + domain)
+    );
+  } else {
```

Comment:
Check warning on line 28 in src/events/messageCreate.ts

GitHub Actions
/ check linting (eslint)

Unnecessary 'else' after 'return'

Reply:
Lint may say unnecessary but typescript gives error bc there is no default return if you remove the else

Reply:
nvm lmao